### DISCUSSION TITLE
How to access cli flags in a plugin?

### DISCUSSION THREAD

**Discussion by hgraca** [composer/composer#11394] [Q&A / Support]
Hi,

Im working on a plugin, and I would like to access the CLI flags that were used when running composer, for example

`composer install --optimize-autoload`

```php
    public static function getSubscribedEvents(): array
    {
        return [
            PackageEvents::POST_PACKAGE_INSTALL => 'onPackageInstall',
        ];
    }

    public function onPackageInstall(PackageEvent $event): void
    {
        $isOptimizeAutoload = ...; // What do I put here?
    }
```

**Comment by Seldaek** [comment]
I don't think you can.. That'd need a reference to the Installer class. Maybe you can listen to the PluginEvents::COMMAND event and check with `$event->getInput()->get('optimize-autoload')` if the command is 'install' or 'update'.. But this may not be a complete picture.

I think the best would be to listen to ScriptEvents::PRE_AUTOLOAD_DUMP and then check `$event->getFlags()['optimize']` (bool). Because doing stuff on package install sounds like the wrong point in time IMO, but I don't know what you're up to.

**Reply by hgraca** [reply]
Cool, I will try that when I have some time and post here the solution I find.

Tkx for taking some time with this. :) 

Btw, what I'm up to is creating a plugin so that we can use composer to manage dependencies on shell packages, the same way we do with PHP packages. Fun stuff! :D

**Reply by hgraca** [reply]
I couldn't find a way following up on @Seldaek hint, so I ended up doing something along these lines:

```php
$arguments = $_SERVER['argv'] ?? [];
$isOptimizeAutoload = in_array('--optimize-autoloader', $arguments, true);
```
